### PR TITLE
Refactor DUCK_STATE to use strong enum

### LINKED ISSUE #12449: Refactor DUCK_STATE to use strong enum

### PR DESCRIPTION
**Pull request by msxuxf**
fixed #12449

### CODE REVIEW

File: src/openrct2/rct2/S6Exporter.cpp [hunk 1]
```diff
@@ -1331,7 +1331,7 @@ void S6Exporter::ExportSpriteMisc(RCT12SpriteBase* cdst, const SpriteBase* csrc)
             dst->frame = src->frame;
             dst->target_x = src->target_x;
             dst->target_y = src->target_y;
-            dst->state = src->state;
+            dst->state = static_cast<uint8_t>(src->state);
```
Comment (tupaschoal) [inline src/openrct2/rct2/S6Exporter.cpp:1334]:
```suggestion
            dst->state = EnumValue(src->state);
```

File: src/openrct2/world/Duck.cpp [hunk 2]
```diff
@@ -20,14 +20,6 @@
 #include <limits>
 
 // clang-format off
```
Comment (tupaschoal) [inline src/openrct2/world/Duck.cpp:22]: Move this to line 24, please

File: src/openrct2/world/Duck.cpp [hunk 3]
```diff
@@ -284,10 +276,10 @@ void Duck::UpdateFlyAway()
 uint32_t Duck::GetFrameImage(int32_t direction) const
 {
     uint32_t imageId = 0;
-    if (state < DUCK_MAX_STATES)
+    if (static_cast<int32_t>(state) < DUCK_MAX_STATES)
```
Comment (tupaschoal) [inline src/openrct2/world/Duck.cpp:279]:
```suggestion
    if (EnumValue(state) < DUCK_MAX_STATES)
```

File: src/openrct2/world/Duck.cpp [hunk 4]
```diff
@@ -284,10 +276,10 @@ void Duck::UpdateFlyAway()
...
     {
         // TODO: Check frame is in range
-        uint8_t imageOffset = DuckAnimations[state][frame];
+        uint8_t imageOffset = DuckAnimations[static_cast<int32_t>(state)][frame];
```
Comment (tupaschoal) [inline src/openrct2/world/Duck.cpp:282]:
```suggestion
        uint8_t imageOffset = DuckAnimations[EnumValue(state)][frame];
```

File: src/openrct2/world/Duck.cpp [hunk 5]
```diff
@@ -333,27 +325,27 @@ void create_duck(const CoordsXY& pos)
...
     duck->MoveTo({ targetPos.x, targetPos.y, 496 });
-    duck->state = DUCK_STATE::FLY_TO_WATER;
+    duck->state = Duck::DuckState::FlyToWater;
     duck->frame = 0;
 }
 
 void Duck::Update()
 {
-    switch (static_cast<DUCK_STATE>(state))
+    switch (static_cast<DuckState>(state))
```
Comment (tupaschoal) [inline src/openrct2/world/Duck.cpp:334]: The cast seems unnecessary, isn't state the enum type already?

Comment (msxuxf) [reply src/openrct2/world/Duck.cpp:334]: Yes. Thank you for reviewing!

File: src/openrct2/world/Duck.cpp (hunk 2 above)
Comment (msxuxf) [reply src/openrct2/world/Duck.cpp]: After moving, the code did not pass the clang-format check.

Comment (tupaschoal) [reply src/openrct2/world/Duck.cpp]: Seems pretty simple to solve, just delete an extra line there

Comment (msxuxf) [reply src/openrct2/world/Duck.cpp]: Thank you, modified.

**Review by tupaschoal — APPROVED** [review]

**Comment by tupaschoal** [comment]
Thanks for the PR. One tip for next time: make sure you mark the conversation/issues opened as resolved when you solve them, so we know you have changed them

**Comment by msxuxf** [comment]
Thank you for merging my PR. And thank you for the tip, will do that next time!

**Pull request merged**